Lesson19/DynamicIntArray.cpp

```diff
@@ -0,0 +1,87 @@
+#include "DynamicIntArray.h"
+#include <iostream>
+
+// Default c-tor
+DynamicIntArray::DynamicIntArray() :m_size(5), m_position(0) {
+    m_elements = new int[m_size];
+}
+
+// C-tor with size
+DynamicIntArray::DynamicIntArray(std::size_t size) : m_size(size), m_position(0) {
+    m_elements = new int[m_size];
+}
+
+// Copy c-tor
+DynamicIntArray::DynamicIntArray(const DynamicIntArray& other) {
+    if (m_size < other.getSize()) {
+        delete[] m_elements;
+        m_size = other.m_size;
+    }
+
+    m_elements = new int[m_size];
+    for (int i = 0; i < m_size; i++) {
+        m_elements[i] = other.m_elements[i];
+    }
+    m_position = other.m_position;
+}
+
+// Free allocated memory
+DynamicIntArray::~DynamicIntArray() {
+    delete[] m_elements;
+}
+
+DynamicIntArray& DynamicIntArray::operator=(const DynamicIntArray& other) {
+    return *this;
+}
```

Comment:
it doesn't work well.
you should remove your data, and allocate memory for a new array to store other.m_elements

Reply:
redone.